app/src/components/Sidebar.tsx

```diff
@@ -21,35 +21,62 @@ class Sidebar extends EnhancedComponent<ISidebarProps, ISidebarState> {
         {genre: "Rap", icon: "https://img.icons8.com/ios-glyphs/30/000000/country-music.png"},
     ]
 
+    private shownGenres: ISidebarGenreChannel[] = [];
+
     protected constructor(props: ISidebarProps) {
         super(props);
         this.state = {
-            searchValue: "",
             genre: "",
             icon: "",
+            shownGenres: [
+                {genre: "Electronic", icon: "https://img.icons8.com/ios-glyphs/30/000000/electronic-music.png"},
+                {genre: "Rock", icon: "https://img.icons8.com/ios-glyphs/30/000000/rock-music.png"},
+                {genre: "Lo-Fi", icon: "https://img.icons8.com/ios-glyphs/30/000000/easy-listening.png"},
+                {genre: "Reggae", icon: "https://img.icons8.com/ios-glyphs/30/000000/reggae.png"},
+                {genre: "Country", icon: "https://img.icons8.com/ios-glyphs/30/000000/country-music.png"},
+                {genre: "Hip-Hop", icon: "https://img.icons8.com/ios-glyphs/30/000000/hip-hop-music.png"},
+                {genre: "Jazz", icon: "https://img.icons8.com/ios-glyphs/30/000000/saxophone.png"},
+                {genre: "Rap", icon: "https://img.icons8.com/ios-glyphs/30/000000/country-music.png"},
+            ],
         };
+        this.onSearch = this.onSearch.bind(this);
     }
 
     private onSearch(event: React.SyntheticEvent) {
+        event.preventDefault();
         let target = event.currentTarget as HTMLInputElement;
-        let value = target.value;
-        this.setState({
-            searchValue: value
-        });
+        let searchValue = target.getElementsByClassName("search_input")[0].getAttribute("value");
+        if (searchValue !== "") {
+            for (let i: number = 0; i < this.musicGenres.length; i++) {
+                if (this.musicGenres[i].genre.includes(searchValue)) {
+                    this.shownGenres.push(this.musicGenres[i]);
+                }
+            }
+            this.setState({
+                ...this.state,
+                shownGenres: this.shownGenres,
+            });
+        } else {
+            this.setState({
+                ...this.state,
+                shownGenres: this.musicGenres,
+            })
+        }
     }
 
     public render() {
         let placeholder = "Search...";
         return (
             <div className="sidebar">
                 <div className="sidebar-search-wrapper">
-                    <TextInput
+                    <SearchBar
                         defaultText={placeholder}
+                        submit={this.onSearch}
```

Comment:
We should search with onUpdate, so as text is entered, the onSearch method is called. Right now, you need to hit the enter button to search